src/store/modules/ui-store.js

```diff
@@ -0,0 +1,30 @@
+export default {
+  // TO-DO: Make ui-stores inheritance in themes from default ui-store like pages/components
+  state: {
+    sidebar: false,
+    microcart: false,
+    overlay: false
+  },
+  getters: {
+    ui (state) {
+      return {
+        sidebar: state.sidebar,
+        microcart: state.microcart,
+        overlay: state.overlay
+      }
+    }
+  },
+  mutations: {
+    setOverlay (state, action) {
```

Comment:
I'm not sure about connecting UI elements like Microcart & Overlay in core.
 Core ui-store should provide the most basic interactions. 
Also i think it should be possible to inherit from this store in `themes/` folder and use theme's ui-store to meet theme's specific requirements (location should be `themes/{theme_name}/store/ui-store.js` so it'll be clear that it's inherited from core `store` folder  )

Can you do this?

Reply:
Sure, so how you want to inherit stores ? by store.registerModule in specific theme ?